docs/design.md

```diff
@@ -91,97 +107,204 @@ This pattern is inspired by [sickle_ui](https://github.com/UmbraLuminosa/sickle_
+By using strings as keys, you can dynamically load assets based on input data such as a level file.
+If you prefer a purely static approach, you can also use an `enum YourAssetHandleKey` and `impl AsRef<str> for YourAssetHandleKey`.
+You can also mix the dynamic and static approach according to your needs.
 
-## Spawning
+## Spawn Command
 
 ### Pattern
 
-Spawn a game object by using an observer:
+Spawn a game object by using a custom command. Inside the command,
+run the spawning code with `world.run_system_once` or  `world.run_system_once_with`:
 
 ```rust
 // monster.rs
-use bevy::prelude::*;
 
-pub(super) fn plugin(app: &mut App) {
-    app.observe(on_spawn_monster);
+#[derive(Debug)]
+pub struct SpawnMonster {
+    pub health: u32,
+    pub transform: Transform,
 }
 
-#[derive(Event, Debug)]
-pub struct SpawnMonster;
+impl Command for SpawnMonster {
+    fn apply(self, world: &mut World) {
+        world.run_system_once_with(self, spawn_monster);
+    }
+}
 
-fn on_spawn_monster(
-    _trigger: Trigger<SpawnMonster>,
+fn spawn_monster(
+    spawn_monster: In<SpawnMonster>,
     mut commands: Commands,
 ) {
     commands.spawn((
         Name::new("Monster"),
+        Health::new(spawn_monster.health),
+        SpatialBundle::from_transform(spawn_monster.transform),
         // other components
     ));
 }
 ```
 
-And then, somewhere else in your code, trigger the observer:
+And then, somewhere else in your code, add the command to the command queue:
 
 ```rust
+// dangerous_forest.rs
+
 fn spawn_monster(mut commands: Commands) {
-    commands.trigger(SpawnMonster);
+    commands.add(SpawnMonster {
+        health: 100,
+        transform: Transform::from_xyz(10.0, 0.0, 0.0),
+    });
 }
 ```
 
 ### Reasoning
 
 By encapsulating the spawning of a game object in a function,
 you save on boilerplate code and can easily change the behavior of spawning.
-An observer is an elegant way to then trigger this function from anywhere in your code.
-A limitation of this approach is that calling code cannot extend the spawn call with additional components or children.
-If you know about a better pattern, please let us know!
+A custom command is an elegant way to then indirectly call this function from anywhere in your code.
+
+A limitation of this approach is that calling code cannot extend the spawn call with additional components or children,
+as custom commands don't return `Entity` or `EntityCommands`. This kind of usage will be possible in future Bevy versions.
+
+## Interaction Callback
+
+### Pattern
+
+When adding a component that can have an interaction, such as a button that can be pressed,
+register a [one-shot system](https://bevyengine.org/news/bevy-0-12/#one-shot-systems) to handle the interaction.
+Add the resulting `SystemId` through a newtype component to your entity:
+
+```rust
+#[derive(Component, Debug, Reflect, Deref, DerefMut)]
+#[reflect(Component, from_reflect = false)]
+pub struct OnPress(#[reflect(ignore)] pub SystemId);
```

Comment:
Does this definition have to be included here? This is provided by the template, not something the user will have to write themselves.

Reply:
Valid question. I added it because:
- it is an integral part of the pattern, even when not part that the user will touch.
- as this document is less of a "tutorial" and more of an explanation of things you might not have seen before, it would make sense to add this as context
- it helps with understanding the "big picture" of the code, so you don't have to ask yourself what that magic `OnPress` thingy is
- the user might want to create their own `OnSomething`s. For example, I can imagine such a callback on a slider.

Reply:
I'd rather link to the code in the template than duplicate it here.

Reply:
[`0d6a2de` (#241)](https://github.com/TheBevyFlock/bevy_quickstart/pull/241/commits/0d6a2de31189ad7565b18b0ddc25d288d3476ff6)